internal/provider/resource_mackerel_role_metadata.go

```diff
@@ -0,0 +1,171 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/mackerelio-labs/terraform-provider-mackerel/internal/mackerel"
+)
+
+var (
+	_ resource.Resource                = (*mackerelRoleMetadataResource)(nil)
+	_ resource.ResourceWithConfigure   = (*mackerelRoleMetadataResource)(nil)
+	_ resource.ResourceWithImportState = (*mackerelRoleMetadataResource)(nil)
+)
+
+func NewMackerelRoleMetadataResource() resource.Resource {
+	return &mackerelRoleMetadataResource{}
+}
+
+type mackerelRoleMetadataResource struct {
+	Client *mackerel.Client
+}
+
+func (r *mackerelRoleMetadataResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_role_metadata"
+}
+
+func (r *mackerelRoleMetadataResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "This resource creates and manages a Role Metadata.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(), // immutable
+				},
+			},
+			"service": schema.StringAttribute{
+				Description: "The name of the service.",
+				Required:    true,
+				Validators: []validator.String{
+					mackerel.ServiceNameValidator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // force new
+				},
+			},
+			"role": schema.StringAttribute{
+				Description: "The name of the role.",
+				Required:    true,
+				Validators: []validator.String{
+					mackerel.RoleNameValidator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // force new
+				},
+			},
+			"namespace": schema.StringAttribute{
+				Description: "The identifier for the metadata.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // force new
+				},
+			},
+			"metadata_json": schema.StringAttribute{
+				Description: "The arbitrary JSON data for the role.",
+				Required:    true,
+				CustomType:  jsontypes.NormalizedType{},
+			},
+		},
```

Comment:
Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/576d3ad25c0da0f9bc19235d86b8e5906ea302a6/mackerel/resource_mackerel_role_metadata.go?plain=1#L24-L43

Reply:
[memo] The `service` and `role` on the original schema does not have `ForceNew: true` (equivalence of `planmodifier.RequiresReplace()`). It seems like a bug (FYI: https://github.com/mackerelio-labs/terraform-provider-mackerel/issues/231).